### PR TITLE
Load sdp from template and add load & sae to sdp

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Utility.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.SchemaDrivenProcessor
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Dynamic;
+
+    public static class DynamicConverter
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Utility.cs]: >DynamicConverter [](start = 24, length = 16)

rename the file name from utility to DynamicConverter?

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Utility.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            var obj = new ExpandoObject();
+            foreach (var pair in value)
+            {
+                var key = pair.Key as string;
+                if (key == null)
+                {
+                    throw new NotSupportedException("Only string key is supported.");
+                }
+                ((IDictionary<String, Object>)obj).Add(key, Convert(pair.Value));
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Utility.cs]: >String [](start = 30, length = 6)

lowecase string?

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/SchemaDrivenDocumentProcessor.cs [hunk 3]
```diff
@@ -7,56 +7,158 @@ namespace Microsoft.DocAsCode.Build.SchemaDrivenProcessor
...
+            {
+                if (assembly != null)
+                {
+                    configuration.WithAssembly(assembly);
+                }
+            }
+
+            try
+            {
+                var container = configuration.CreateContainer();
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/SchemaDrivenDocumentProcessor.cs]: >configuration.CreateContainer(); [](start = 32, length = 32)

reuse CompositionContainer.GetContainer, then getexports?

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        private const string SchemaFileEnding = ".schema.json";
+        public static readonly ThreadLocal<JsonSerializer> DefaultSerializer = new ThreadLocal<JsonSerializer>(
+               () => new JsonSerializer
+               {
+                   NullValueHandling = NullValueHandling.Ignore,
+                   ReferenceLoopHandling = ReferenceLoopHandling.Serialize,
+                   ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                   Converters =
+                   {
+                     new StringEnumConverter { CamelCaseText = true },
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs]: >   [](start = 19, length = 2)

format, remove emply lines

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        public static DSchema Load(string schemaPath)
+        {
+            if (string.IsNullOrEmpty(schemaPath))
+            {
+                throw new ArgumentNullException(nameof(schemaPath));
+            }
+            if (!schemaPath.EndsWith(SchemaFileEnding, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidSchemaException($"Schema path must be end with {SchemaFileEnding}");
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs]: >be [](start = 68, length = 2)

remove

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/PropertySchema.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.SchemaDrivenProcessor
+{
+    using System.Collections.Generic;
+
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+
+    public class BaseSchema
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/PropertySchema.cs]: >BaseSchema [](start = 17, length = 10)

rename filename

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+    using Newtonsoft.Json.Serialization;
+
+    using Microsoft.DocAsCode.Exceptions;
+
+    public class DSchema : BaseSchema
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs]: >DSchema [](start = 17, length = 7)

naming

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+                if (schema.Type != JSchemaType.Object)
+                {
+                    throw new InvalidSchemaException("Type for the root schema object must be object");
+                }
+
+                return schema;
+            }
+        }
+
+        public static DSchema Load(string schemaPath)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DSchema.cs]: >schemaPath [](start = 42, length = 10)

what about ".schema.json" self?

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/PropertySchema.cs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        public Dictionary<string, BaseSchema> Properties { get; set; }
+        public BaseSchema Items { get; set; }
+        public ReferenceType Reference { get; set; }
+        public ContentType ContentType { get; set; }
+
+        public List<string> Tags { get; set; }
+
+        public MergeType MergeType { get; set; }
+
+    }
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/PropertySchema.cs]: minor: remove extra line

File: src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DocumentSchema.cs [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+            if (!schemaPath.EndsWith(SchemaFileEnding, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidSchemaException($"Schema path {schemaPath} does not end with {SchemaFileEnding}");
+            }
+
+            var fileName = Path.GetFileName(schemaPath);
+            var name = fileName.Substring(0, fileName.Length - SchemaFileEnding.Length);
+            if (string.IsNullOrEmpty(name))
+            {
+                {
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDrivenProcessor/Models/DocumentSchema.cs]: remove

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**